### ISSUE TITLE
Resource discovery: Implement resource card

### ISSUE THREAD

**Issue by MisRob** [learningequality/kolibri#12319]
## Blocked by

- https://github.com/learningequality/kolibri-design-system/pull/625 (however, since the PR is nearly ready, can be developed in parallel locally via [Running Kolibri with local Kolibri Design System](https://kolibri-dev.readthedocs.io/en/develop/howtos/development_with_kds.html))

## Summary

In support of coach resource discovery we need to implement a new resource card. This will be one of the first cards in Kolibri built on top of `KCard`.

The final card is expected to look like:

![Screenshot from 2024-06-18 20-41-15](https://github.com/learningequality/kolibri/assets/13509191/30b31c64-3ebf-4f96-82d4-c3bb198ecac2)

It needs to show

- Resource title
- Resource description
- Metadata
- Resource thumbnail picture
- Resource thumbnail placeholder when picture is not available. The placeholder icon corresponds to learning activity type of the resource.
- Bookmark button icon (only emits event on click)
- Info button (only emits event on click)

## References

- [Resource discovery designs (see resource cards)](https://www.figma.com/design/N9nfriAlR3LoMWNuasaljA/Coach-report-metadata?node-id=1630-36418&t)
- [Card specs designs](https://www.figma.com/design/1tQQ0Sw1yDjzO66hyjLVPM/Kolibri-UX-Fixes?node-id=1868-55351&t=WoKMRF2V5JIeRB8b-0)
- https://github.com/learningequality/kolibri/issues/11888
- Sibling cards: https://github.com/learningequality/kolibri/issues/12317, https://github.com/learningequality/kolibri/issues/12318

## Guidance

- Create a completely new component `<AccessibleResourceCard>`  and use `<KCard>` at its basis
- Regarding the structure of production data that the card should display, examine existing resource cards in the Library and also see `useSearch` composable. In the scope of this issue, assume that data for the new card will have the same structure.
- For the purpose of development and review, display the new card at any place in Kolibri and remove the preview shortly before the final merge. It may be handy to work on the Library page because it has access to all data.
- Even though any of the existing cards shouldn’t be changed, they may contain some useful pieces of code for the new card so it’s recommended to examine them (see `ResourceCard.vue` when `contentNode.is_leaf` is truthy, `LearningActivityIcon.vue`, `LearningActivityLabel.vue`, ...)
- It is possible that `KCard` will need further updates to support all Kolibri cards. If there’s something that `KCard` doesn’t support yet, discuss with relevant people and determine next steps.

## Out of scope

- Metadata pills
- Anything grid related (this includes card’s min and max width which will be grid’s responsibility, as well as checkboxes)
- Refactoring existing card components in Kolibri
- Implementing other parts of the coach resource discovery such as the side panel where cards will be displayed
- ‘Bookmark’ and ‘info’ fully functioning

## Acceptance criteria

- [x]  The card is built on top of the `KCard` and contains elements listed above
- [x]  It supports all possible modes (e.g. for all various learning activities that a resource can have)
- [x]  It follows the designs closely (or discrepancies get discussed)
- [x]  It content tolerant (renders well for short strings, long strings, missing information, missing thumbnail picture, etc.)
- [x]  It is responsive
- [x]  It supports RTL (note that `KCard` [will be responsible for basic layout RTL support](https://github.com/learningequality/kolibri-design-system/issues/665), but some additional tweaks may be needed for content that is provided to it via slots etc.)
- [x]  It is ready to receive production data, meaning it doesn’t use hardcoded data inside the card but rather receives data from outside like it would in the final feature, for example in the form of an object passed via prop (or any other relevant manner)
- [x]  For the purpose of preview and QA, please include at least a couple of `<AccessibleResourceCard>`'s in `<ul>` and feed them with data that have same structure as production data and enough variance (see content tolerance criteria). Demonstrate all learning activities.
- [x] If some components used in the current cards are re-used in this new card and tweaked for the purpose, there are no regressions in the old cards

**Comment by MisRob** [comment]
Metadata are out of scope of this issue

**Comment by Spoorthy1423** [comment]
Hey!! I would like to work on this issue @MisRob  please assign me this issue!!

**Comment by MisRob** [comment]
Hi @Spoorthy1423, thanks for volunteering. This issue is not available for assignment.

There are contributing opportunities in three repositories. See the contributing guidelines including links to issues suitable for contribution for each repository below. **I would ask you to read the contributing guidelines at first and volunteer for one issue at a time.**

- [Kolibri Contributing Guidelines](https://github.com/learningequality/kolibri/blob/release-v0.16.x/CONTRIBUTING.md)
- [Kolibri Design System Contributing Guidelines](https://github.com/learningequality/kolibri-design-system/blob/main/CONTRIBUTING.md)
- [Studio Contributing Guidelines](https://github.com/learningequality/studio/blob/unstable/CONTRIBUTING.md)

You can also see the list of unassigned ["help wanted"](https://github.com/search?q=org%3Alearningequality+is%3Aopen+is%3Aissue+label%3A%22help+wanted%22+no%3Aassignee&type=issues) or ["good first issue"](https://github.com/search?q=org%3Alearningequality+is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22+no%3Aassignee&type=issues) issues across all repositories.

**Comment by MisRob** [comment]
WIP in https://github.com/learningequality/kolibri/pull/12418/

**Comment by AllanOXDi** [comment]
completed in https://github.com/learningequality/kolibri/pull/12418
